pow/src/test/java/tech/pegasys/artemis/pow/Eth1DataManagerTest.java

```diff
@@ -0,0 +1,381 @@
+            new MockBlock(10, 132),
+            new MockBlock(11, 138),
+            // Cache Range Lower Bound: 142
+            new MockBlock(12, 144),
+            new MockBlock(13, 150),
+            new MockBlock(14, 156),
+            new MockBlock(15, 162),
+            new MockBlock(16, 168),
+            new MockBlock(17, 174),
+            // Cache Range Upper Bound: 177
+            new MockBlock(18, 180),
+            new MockBlock(19, 186));
+
+    MockBlock latestBlockRequest = new MockBlock(21, 198);
+
+    setupWeb3jMockedBlockResponses(eth1Blocks, latestBlockRequest);
+
+    eth1DataManager.start();
+
+    assertThat(eventCapture.getEth1BlockEvents().size()).isEqualTo(8);
+
+    List<Integer> eth1BlockTimestamps =
+        eventCapture.getEth1BlockEvents().stream()
+            .filter(event -> event.getClass().equals(CacheEth1BlockEvent.class))
+            .map(CacheEth1BlockEvent::getBlockTimestamp)
+            .map(UnsignedLong::intValue)
+            .collect(Collectors.toList());
+
+    assertThat(eth1BlockTimestamps)
+        .containsExactlyInAnyOrder(138, 144, 150, 156, 162, 168, 174, 180);
+  }
+
+  @Test
+  void cacheStartup_recalculateSecondsToFindMidRangeBlock() {
+    // There is ice-age! Average block times are at about 20 seconds! What does Teku do? FUNCTION.
+    List<MockBlock> eth1Blocks =
+        Arrays.asList(
+            new MockBlock(15, 78),
+            new MockBlock(16, 98),
+            new MockBlock(17, 118),
+            new MockBlock(18, 138),
+            // Cache Range Lower Bound: 142
+            new MockBlock(19, 158),
+            // Cache Range Upper Bound: 177
+            new MockBlock(20, 178));
+
+    MockBlock latestBlockRequest = new MockBlock(21, 198);
+
+    setupWeb3jMockedBlockResponses(eth1Blocks, latestBlockRequest);
+
+    eth1DataManager.start();
+
+    assertThat(eventCapture.getEth1BlockEvents().size()).isEqualTo(3);
+
+    List<Integer> eth1BlockTimestamps =
+        eventCapture.getEth1BlockEvents().stream()
+            .filter(event -> event.getClass().equals(CacheEth1BlockEvent.class))
+            .map(CacheEth1BlockEvent::getBlockTimestamp)
+            .map(UnsignedLong::intValue)
+            .collect(Collectors.toList());
+
+    assertThat(eth1BlockTimestamps).containsExactlyInAnyOrder(158, 138, 178);
+  }
+
+  @Test
+  void cacheStartup_retryStartup() {
+    Request mockRequest = mockFailedRequest();
+    when(web3j.ethGetBlockByNumber(any(), eq(false))).thenReturn(mockRequest);
+
+    eth1DataManager.start();
+
+    verify(web3j, times(Math.toIntExact(Constants.ETH1_CACHE_STARTUP_RETRY_GIVEUP)))
+        .ethGetBlockByNumber(any(), eq(false));
+  }
+
+  @Test
+  void onTick_startupNotDone() {
+    eventBus = spy(new EventBus());
+    eth1DataManager =
+        spy(
+            new Eth1DataManager(
+                web3j, eventBus, depositContractListener, asyncRunner, timeProvider));
+    eventBus.post(new Date());
+    verifyNoInteractions(eth1DataManager);
```

Comment:
I'd avoid spies here and just assert that there's no registration on the event bus until start is called:

```
eventBus = mock(EventBus.class);
eth1DataManager = new Eth1DataManager(web3j, eventBus, depositContractListener, asyncRunner, timeProvider);
verifyZeroInteractions(eventBus);
```

Both approaches are a bit specific to the implementation but at least this one will catch all potential interactions, even if `Eth1DataManager` switches to to a different event type.

Reply:
Makes sense. Tests are less brittle that way. Done.